app/src/main/java/com/github/studydistractor/sdp/createEvent/CreateEventServiceFirebase.kt

```diff
@@ -0,0 +1,28 @@
+package com.github.studydistractor.sdp.createEvent
+
+import com.github.studydistractor.sdp.data.FirebaseChat
+import com.github.studydistractor.sdp.data.FirebaseEvent
+import com.google.android.gms.tasks.Task
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+
+class CreateEventServiceFirebase : CreateEventModel {
+    private val eventsDB: DatabaseReference = FirebaseDatabase.getInstance().getReference("Events")
+    private val chatDB: DatabaseReference = FirebaseDatabase.getInstance().getReference("ChatEvent")
+    override fun createEvent(eventInformation: Map<String, Any?>): Task<Void> {
```

Comment:
Why is this not already taking in an "Event" instance?

Reply:
Yeah! Imo we should make both uids nullable in the Event dataclass instead. This is what I've seen done in other codebases.

Doing it with a map is not very DRY, and making uid's nullable makes more sense imo

Reply:
I think that we have agreed that an event should always have an event id and a chat id. If it doesn't, it's not an event. As a result, I think it's more clear, semantically speaking, that we only pass the event info (because that's what we do) instead of a "fake" event that is not really an event but will become an event once we add the event id and chat id.

Reply:
I think having the uid fields nullable or, better even, *optional* is the best solution here for DRY and coupling the Event and "eventInformation" fields. It's also what I've seen done in other codebases.

If you want to have a separate EventInformation class that holds everything except the uids, and have an Event class that inherits from it but adds the uids I think it's fine as well, but we really need to couple the eventInformation variable and the Event data class as much as possible imo.

Not coupling them means you have this list of strings that you are rewriting in both the viewmodel and the service(s), which is not good for IDE integration, and forces you to write a lot of code that can be avoided.